### PR TITLE
Better JSON and Error Types

### PR DESCRIPTION
**Pull request by raayan**

### CODE REVIEW

File: core/src/lib.rs [hunk 1]
```diff
@@ -112,24 +108,71 @@ pub type DefaultFileService = FileServiceImpl<
     DefaultFileEncryptionService,
 >;
 
+macro_rules! impl_get_variants {
```
Comment (Parth) [inline core/src/lib.rs:111]: Maybe worth dropping some comments here explaining what's going on. Also let's move this to the bottom of this file.

File: core/src/lib.rs [hunk 2]
```diff
@@ -112,24 +108,71 @@ pub type DefaultFileService = FileServiceImpl<
...
+}
+
+#[derive(Debug, Serialize, EnumIter)]
 pub enum InitLoggerError {
-    UnexpectedError(String),
+    Stub, // TODO: Enums should not be empty
 }
 
-pub fn init_logger(log_path: &Path) -> Result<(), InitLoggerError> {
+pub fn init_logger(log_path: &Path) -> Result<(), Error<InitLoggerError>> {
```
Comment (Parth) [inline core/src/lib.rs:163]: Could this be an `Error<()>`?

File: core/src/lib.rs (hunk 1 above)
Comment (raayan) [reply core/src/lib.rs:111]: sure thing

File: core/src/lib.rs (hunk 2 above)
Comment (raayan) [reply core/src/lib.rs:163]: yes we can

**Pull request merged**